### PR TITLE
✨ Add `--local` option to CLI

### PR DESCRIPTION
**Pull request by laurentsimon**
* **Please check if the PR fulfills these requirements**
- [] Tests for the changes have been added (for bug fixes / features)
- [X] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

This add `--local` option as argument. No breaking changes.

### CODE REVIEW

File: clients/localdir/client.go [hunk 1]
```diff
@@ -116,10 +136,9 @@ func listFiles(clientPath string, predicate func(string) (bool, error)) ([]strin
 // ListFiles implements RepoClient.ListFiles.
 func (client *localDirClient) ListFiles(predicate func(string) (bool, error)) ([]string, error) {
 	client.once.Do(func() {
-		client.files, client.errFiles = listFiles(client.path, predicate)
+		client.files, client.errFiles = listFiles(client.path)
 	})
-
-	return client.files, client.errFiles
+	return applyPredicate(client.path, client.files, client.errFiles, predicate)
```
Comment (azeemshaikh38) [inline clients/localdir/client.go:141]: Consider making this a struct fn so you don't have to pass all the `client.*` objects.

File: clients/localdir/client.go [hunk 2]
```diff
@@ -92,22 +98,36 @@ func listFiles(clientPath string, predicate func(string) (bool, error)) ([]strin
...
+func applyPredicate(clientPath string, clientFiles []string,
+	errFiles error, predicate func(string) (bool, error)) ([]string, error) {
+	if errFiles != nil {
+		return nil, errFiles
+	}
+
+	files := []string{}
+	for _, pathfn := range clientFiles {
+		// Remove prefix of the folder.
+		p := trimPrefix(pathfn, clientPath)
```
Comment (azeemshaikh38) [inline clients/localdir/client.go:122]: Save the trimmed path to `client.files` to avoid running `trimPrefix` everytime?

**Review by azeemshaikh38 — APPROVED** [review]

File: clients/localdir/client.go (hunk 2 above)
Comment (laurentsimon) [reply clients/localdir/client.go]: just realized the call was only needed once in the `do.Once`. My bad. So I removed the call entirely here.

File: clients/localdir/client.go (hunk 1 above)
Comment (laurentsimon) [reply clients/localdir/client.go]: I removed `client.path` as it was not needed, so it takes only 3 arguments.

**Pull request merged**